benchmark/sampler/hetero_neighbor.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+import ast
+import os.path as osp
+import time
+
+import torch
+import torch_geometric.transforms as T
+import torch_sparse  # noqa
+from torch_geometric.datasets import OGB_MAG
+from torch_geometric.loader.utils import get_input_nodes
+from torch_geometric.sampler.utils import remap_keys, to_hetero_csc
+from tqdm import tqdm
+
+import pyg_lib
+from pyg_lib.testing import withSeed
+
+argparser = argparse.ArgumentParser('Hetero neighbor sample benchmark')
+argparser.add_argument('--batch-sizes', nargs='+', type=int, default=[
+    512,
+    1024,
+    2048,
+    4096,
+    8192,
+])
+argparser.add_argument('--num_neighbors', type=ast.literal_eval, default=[
+    [-1],
+    [15, 10, 5],
+    [20, 15, 10],
+])
+argparser.add_argument('--replace', action='store_true')
+argparser.add_argument('--shuffle', action='store_true')
+argparser.add_argument('--do_train', action='store_true')
+# TODO (kgajdamo): Support undirected hetero graphs
+# argparser.add_argument('--directed', action='store_true')
+# TODO (kgajdamo): Enable CSR
+# argparser.add_argument('--csc', action='store_true')
+
+args = argparser.parse_args()
+
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+path = osp.join(osp.dirname(osp.realpath(__file__)), '../../data/OGB')
+transform = T.ToUndirected(merge=True)
+dataset = OGB_MAG(path, preprocess='metapath2vec', transform=transform)
```

Comment:
We can move this to `pyg_lib.testing.withDataset`. WDYT? And then have it return a dictionary of `(rowptr, col)` entries?

Reply:
Ok, I'll try

Reply:
Thanks! Can you clarify what is not working for sampling with replacement? Does it simply crash?

Reply:
I crashes in uniform_sample() -> add()

Reply:
Thanks, will take a look!